pubsublite/spark-connector/spark_streaming_test.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import pathlib
+import re
+from typing import Generator
+import uuid
+
+from google.api_core.exceptions import NotFound
+from google.cloud import dataproc_v1, storage
+from google.cloud.pubsublite import AdminClient, Subscription, Topic
+from google.cloud.pubsublite.types import (
+    CloudRegion,
+    CloudZone,
+    SubscriptionPath,
+    TopicPath,
+)
+import pytest
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+PROJECT_NUMBER = os.environ["GOOGLE_CLOUD_PROJECT_NUMBER"]
+CLOUD_REGION = "us-west1"
+ZONE_ID = "a"
+CLUSTER_ID = os.environ["PUBSUBLITE_CLUSTER_ID"]
+BUCKET = os.environ["PUBSUBLITE_BUCKET_ID"]
+UUID = uuid.uuid4().hex
+TOPIC_ID = "spark-streaming-topic-" + UUID
+SUBSCRIPTION_ID = "spark-streaming-subscription-" + UUID
+CURRENT_DIR = pathlib.Path(__file__).parent.resolve()
+
+
+@pytest.fixture(scope="module")
+def client() -> Generator[AdminClient, None, None]:
+    yield AdminClient(CLOUD_REGION)
+
+
+@pytest.fixture(scope="module")
+def topic(client: AdminClient) -> Generator[Topic, None, None]:
+    location = CloudZone(CloudRegion(CLOUD_REGION), ZONE_ID)
+    topic_path = TopicPath(PROJECT_NUMBER, location, TOPIC_ID)
+    topic = Topic(
+        name=str(topic_path),
+        partition_config=Topic.PartitionConfig(
+            # A topic must have at least one partition.
+            count=1,
+            capacity=Topic.PartitionConfig.Capacity(
+                # Set publish throughput capacity per partition to 4 MiB/s.
+                publish_mib_per_sec=4,
+                # Set subscribe throughput capacity per partition to 4 MiB/s.
+                subscribe_mib_per_sec=8,
+            ),
+        ),
+        retention_config=Topic.RetentionConfig(
+            per_partition_bytes=30 * 1024 * 1024 * 1024,
+        ),
+    )
+
+    try:
+        response = client.get_topic(topic.name)
+    except NotFound:
+        response = client.create_topic(topic)
+
+    yield response
+
+    try:
+        client.delete_topic(response.name)
+    except NotFound:
+        pass
```

Comment:
Please add a print statement here for easier test debugging

Reply:
The error message for APIExceptions should be self-explanatory. It will print out the fully qualified resource name plus the message "is not found".

Reply:
I see what you're saying - yep - the service will show that in Cloud Logging - in general though, we prefer not to introduce the "except/pass" pattern in samples because I think it's not a pattern that's generally recommended in Python - could you surface what is printed in Cloud Logs?

something like
```
except NotFound as e:
    print(e) # or print(e.message) - whatever you need to get the useful part across for maintainers/debugging
```